Makefile

```diff
@@ -92,6 +92,14 @@ unittest: ## Run unittests on the Helm chart
 	@echo "Running unit tests on helm chart..."
 	cd helm-charts/splunk-otel-collector && helm unittest --strict -f "../../test/unittests/*.yaml" . || exit 1
 
+# Example Usage:
+#   make functionaltest
+#   make functionaltest SKIP_SETUP=true SKIP_TEARDOWN=true SKIP_TESTS=true TEARDOWN_BEFORE_SETUP=true TAGS="integration" UPDATE_EXPECTED_RESULTS=true KUBE_TEST_ENV="kind" KUBECONFIG="/path/to/kubeconfig"
+.PHONY: functionaltest
+functionaltest: ## Run functional tests for this Helm chart with optional tags and environment variables
+	@echo "Running functional tests for this helm chart..."
+	cd functional_tests && go mod tidy && go test -v $(if $(TAGS),-tags $(TAGS)) || exit 1
```

Comment:
go mod tidy?

Reply:
I'm seeing if this is necessary. So far I couldn't get caching to work properly without a go.sum file generated by go mod tidy. I might check in a a go.sum file for the functional tests. Taking advice if you have any.

Reply:
Updating go.sum is typically done via update, go.sum contains cryptographic proofs of the artifacts used by go.mod are up to date. You would tidy and merge to main changes to make sure nothing nefarious runs by inadvertence.

Reply:
Is `|| exit 1` necessary?

Reply:
We should have a go.sum already. My understanding was that the github action was using the hash of our checked in go.sum to track if it could reuse from a previous run.

Reply:
Yeah it turns out my local git cache showed the go.sum as ignored or non-existent which was confusing for me and my IDE. A cache clear helped that out. I was able to remove the go mod tidy here. 

I've had issues with not adding `|| exit 1` in this specific makefile before and the rest of this file uses `|| exit 1`. I think we can likely refactor removing `|| exit 1` in the future in a separate PR.